ext/storage_sources/s3_store/aws_bucket_conn.cpp

```diff
@@ -0,0 +1,124 @@
+#include <aws/core/Aws.h>
+#include <aws/s3-crt/S3CrtClient.h>
+#include <iostream>
+#include <fstream>
+
+#include "aws_bucket_conn.h"
+
+#include <aws/s3-crt/model/DeleteObjectRequest.h>
+#include <aws/s3-crt/model/ListObjectsRequest.h>
+#include <aws/s3-crt/model/PutObjectRequest.h>
+
+#include <aws/s3-crt/model/Object.h>
+#include <vector>
+#include <string>
+#include <sys/stat.h>
+
+bool
+aws_bucket_conn::list_buckets(Aws::Vector<Aws::S3Crt::Model::Bucket> *buckets) const
+{
+    Aws::S3Crt::Model::ListBucketsOutcome outcome = m_s3_crt_client.ListBuckets();
+
+    if (outcome.IsSuccess()) {
+        *buckets = outcome.GetResult().GetBuckets();
+        return true;
+    } else {
+        std::cout << "Error in list_buckets: " << outcome.GetError().GetMessage() << std::endl
+                  << std::endl;
+        return false;
+    }
+}
+
+bool
+aws_bucket_conn::list_bucket_objects(
+  const Aws::String bucket_name, Aws::Vector<Aws::S3Crt::Model::Object> *bucket_objects) const
+{
+    Aws::S3Crt::Model::ListObjectsRequest request;
+    request.WithBucket(bucket_name);
+    Aws::S3Crt::Model::ListObjectsOutcome outcomes = m_s3_crt_client.ListObjects(request);
+
+    if (outcomes.IsSuccess()) {
+        *bucket_objects = outcomes.GetResult().GetContents();
+        return true;
+    } else {
+        std::cout << "Error in list_bucket_objects: " << outcomes.GetError().GetMessage()
+                  << std::endl;
+        return false;
+    }
+}
+
```

Comment:
Please put a function header. Also, please check if we can update s_all to complaint if function headers are missing.

Reply:
@jiechenbo, what was the script you worked on that checked headers?

Reply:
Hmm the script i worked on is **function.py**, the script only checks missing headers of c files and does not check cpp files. Should we allow that too as well?